src/java.base/share/classes/jdk/internal/io/JdkConsoleImpl.java

```diff
@@ -119,8 +119,17 @@ public char[] readPassword(String fmt, Object ... args) {
                         else
                             ioe.addSuppressed(x);
                     }
-                    if (ioe != null)
+                    if (ioe != null) {
```

Comment:
I assume it's only important to zero the buffer when restoring the echo setting fails. If I read the changes correctly then it's been zero's even if readLine fails. It's okay, just probably unnecessary for that case.

Reply:
Yes, but this way it's assured we clean before we return.